### ISSUE TITLE
[Bug]: react/prop-types false positives since upgrading typescript-eslint to v8

### ISSUE THREAD

**Issue by dan-serendipity** [jsx-eslint/eslint-plugin-react#3796]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

Since upgrading `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` to `v8.0.1`, I'm getting false positives for `react/prop-types`. It looks like this might have been an issue [before](https://github.com/jsx-eslint/eslint-plugin-react/issues/3140).

```
const Divulus = forwardRef<HTMLDivElement, PropsWithChildren>(({ children }, ref) => <div ref={ref}>{children}</div>);
Divulus.displayName = 'X';
```

`'children' is missing in props validation eslint(react/prop-types)`

`eslint`

### Expected Behavior

No error.

### eslint-plugin-react version

v7.35.0

### eslint version

v8.57.0

### node version

-

**Comment by ljharb** [comment]
We don’t even support v6 of the ts parser yet, let alone v7 or v8. See #3629.

**Comment by stephane-ruhlmann** [comment]
I encountered the same issue when upgrading to typescript-eslint 8.x

I confirm that avoiding destructuring and renaming the `props` parameter solves the issue.

```typescript
type TProps = {
  title: string;
};

const Section = forwardRef<Ref, TProps>((props, ref) => {
  return <section ref={ref}>{props.title}</section>;
});
```
Output: `❌ 'title' is missing in props validation        react/prop-types`



```typescript
type TProps = {
  title: string;
};

const Section = forwardRef<Ref, TProps>(({ title }, ref) => {
  return <section ref={ref}>{title}</section>;
});
```
Output: `❌ 'title' is missing in props validation        react/prop-types`




```typescript
type TProps = {
  title: string;
};

const Section = forwardRef<Ref, TProps>((myProps, ref) => {
  return <section ref={ref}>{myProps.title}</section>;
});
```
Works ✅

Not ideal but a simple workaround 🤷‍♂️

**Comment by dan-serendipity** [comment]
Fixed in https://github.com/jsx-eslint/eslint-plugin-react/releases/tag/v7.36.0 via https://github.com/jsx-eslint/eslint-plugin-react/pull/3629. Thanks all!
